src/index.tsx

```diff
@@ -83,7 +83,15 @@ export const TagsInput = ({
     const text = e.target.value;
 
     if (e.key === "Backspace" && tags.length && !text) {
-      setTags(tags.slice(0, -1));
+      const tagsCopy = [...tags];
+      const poppedTag = tags.pop();
+      setTags(tagsCopy.slice(0, -1));
+
+      /** TODO: explanation: added an space to keep the text to put
+       * into the input and not remove the last letter.
+       * I am looking for better ways to do this, make it more elegant.
+       * */
+      e.target.value = `${poppedTag} `;
```

Comment:
I guess if possible logic can be improved a bit in first 3 lines something like below?

```js
e.target.value = `${tags.at(-1)} `;

setTags([...tags.slice(0, -1)]);
```

I think slice is probably immutable so we might be able to avoid spread also?

Reply:
This way it's more simple and maybe more elegant, I applying this and add the option by props to do this optional. :)